nexus/src/internal_api/params.rs

```diff
@@ -98,6 +99,55 @@ pub struct DatasetPutResponse {
     pub quota: Option<ByteCount>,
 }
 
+/// Describes the purpose of the service.
+#[derive(Debug, Serialize, Deserialize, JsonSchema, Clone, Copy, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum ServiceKind {
+    InternalDNS,
+    Nexus,
+    Oximeter,
+}
+
+impl fmt::Display for ServiceKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use ServiceKind::*;
+        let s = match self {
+            InternalDNS => "internal_dns",
+            Nexus => "nexus",
+            Oximeter => "oximeter",
+        };
+        write!(f, "{}", s)
+    }
+}
+
+impl FromStr for ServiceKind {
+    type Err = omicron_common::api::external::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        use ServiceKind::*;
+        match s {
+            "nexus" => Ok(Nexus),
+            "oximeter" => Ok(Oximeter),
```

Comment:
This appears to be missing `"internal_dns"` as a match arm.

Reply:
Good catch. The match-on-string cases don't give any help from types!

Reply:
Done.